addons/controllers/packageinstallstatus_controller.go

```diff
@@ -0,0 +1,442 @@
+	}
+
+	r.Log.WithValues("pkgi-name", pkgi.Name).Info("Mapping PackageInstalls to cluster")
+
+	clusterNamespace, clusterName := r.getClusterNamespacedName(pkgi)
+	if clusterNamespace == "" || clusterName == "" {
+		return nil
+	}
+
+	// TODO: check if the following check is needed here considering that we check it in PkgiStatusChanged predicate
+	isManaged, err := r.isPackageManaged(clusterName, clusterNamespace, pkgi.Name)
+	if err != nil || !isManaged {
+		return nil
+	}
+
+	return []ctrl.Request{{NamespacedName: client.ObjectKey{Namespace: clusterNamespace, Name: clusterName}}}
+}
+
+// PkgiStatusChanged returns a predicate.Predicate that filters pkgi objects which their status has changed
+func (r *PackageInstallStatusReconciler) PkgiStatusChanged(log logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return r.processPkgiStatus(e.Object, log.WithValues("predicate", "createEvent"))
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return r.processPkgiStatus(e.ObjectNew, log.WithValues("predicate", "updateEvent"))
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return r.processPkgiStatus(e.Object, log.WithValues("predicate", "deleteEvent"))
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return r.processPkgiStatus(e.Object, log.WithValues("predicate", "genericEvent"))
+		},
+	}
+}
+
+// processPkgiStatus returns true if pkgi status should be processed.
+// pkgi status can be processed if it is a managed package and exists in the list of packages defined in ClusterBootstrap
+func (r *PackageInstallStatusReconciler) processPkgiStatus(o client.Object, log logr.Logger) bool {
+	var pkgi *kapppkgiv1alpha1.PackageInstall
+	switch obj := o.(type) {
+	case *kapppkgiv1alpha1.PackageInstall:
+		pkgi = obj
+	default:
+		log.Info("Expected object type of PackageInstall. Got object type", "actualType", fmt.Sprintf("%T", o))
+		return false
+	}
+
+	clusterNamespace, clusterName := r.getClusterNamespacedName(pkgi)
+	if clusterNamespace == "" || clusterName == "" {
+		return false
+	}
+
+	isManaged, err := r.isPackageManaged(clusterName, clusterNamespace, pkgi.Name)
+	if err != nil {
+		return false
+	}
+
+	return isManaged
+}
+
+// isPackageManaged checks if the provided PackageInstall is among the list of managed(core/additional) packages
+func (r *PackageInstallStatusReconciler) isPackageManaged(clusterName, clusterNamespace, pkgiName string) (bool, error) {
+	clusterBootstrap := &runtanzuv1alpha3.ClusterBootstrap{}
+	clusterBootstrapKey := client.ObjectKey{Namespace: clusterNamespace, Name: clusterName}
+	if err := r.Client.Get(r.Ctx, clusterBootstrapKey, clusterBootstrap); err != nil {
+		r.Log.Error(err, fmt.Sprintf("error getting ClusterBootstrap resource for cluster '%s/%s'", clusterBootstrapKey.Namespace, clusterBootstrapKey.Name))
+		return false, err
+	}
+
+	packages := append([]*runtanzuv1alpha3.ClusterBootstrapPackage{
+		clusterBootstrap.Spec.CNI,
```

Comment:
blocker: we have `CNIs` now

Reply:
No it changed to CNI in [another PR](https://github.com/vmware-tanzu/tanzu-framework/pull/1834)